src/charm.py

```diff
@@ -37,10 +41,34 @@ def __init__(self, *args):
             self.model.unit.status = WaitingStatus("Waiting for leadership")
             return
 
+        self.prometheus_provider = MetricsEndpointProvider(
+            charm=self,
+            relation_name="metrics-endpoint",
+            jobs=[
+                {
+                    "metrics_path": METRICS_PATH,
+                    "static_configs": [
+                        {"targets": ["*:{}".format(METRICS_PORT)]}
+                    ],
+                }
+            ],
+        )
+
+        self.dashboard_provider = GrafanaDashboardProvider(self)
+
         self.framework.observe(self.on.install, self._install)
         self.framework.observe(self.on.remove, self._remove)
         self.framework.observe(self.on.update_status, self._update_status)
 
+        monitoring_events = [
+            self.on["metrics-endpoint"].relation_changed,
```

Comment:
Add `relation_joined`

Reply:
I also believe we have a constant that can be imported from the library itself. @simskij IIRC we even have an event one can register to, no?

Reply:
Actually… I do not understand what is the benefit of registering these hooks. @natalian98